File: Java/DecoratorPattern/src/addons/WhippedCream.java
package addons;

import coffee.Coffee;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 27-05-2021
 * Time: 04:33 AM
 */
public class WhippedCream extends Addons{

    Coffee coffee;

    double whippedCreamCost = 10.0;

    public WhippedCream(Coffee coffee) {
        this.coffee = coffee;
    }

    @Override
    public String getDescription() {
        return coffee.getDescription()+" + Whipped Cream : Rs "+whippedCreamCost;
    }

    @Override
    public double cost() {
        return coffee.cost()+whippedCreamCost;
    }
}

File: Kotlin/DecoratorPattern/src/addons/WhippedCream.kt
package addons

import coffee.Coffee

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 27-05-2021
 *   Time: 04:57 AM
 *
 */
class WhippedCream(c: Coffee): Addons() {
    val coffee = c

    var whippedCreamCost = 3.0

    override fun getCoffeeDescription(): String {
        return "${coffee.getCoffeeDescription()} + Whipped Cream : Rs $whippedCreamCost"
    }

    override fun cost(): Double {
        return coffee.cost()+whippedCreamCost
    }
}